src/bentoml/_internal/utils/formparser.py

```diff
@@ -202,17 +310,21 @@ async def populate_multipart_requests(request: Request) -> t.Dict[str, Request]:
         raise BentoMLException("Invalid multipart requests")
 
     reqs: dict[str, Request] = dict()
-    for field_name, headers, data in form:
+    for field_name, data in form.items():
         scope = dict(request.scope)
-        ori_headers = dict(scope.get("headers", dict()))
-        ori_headers = t.cast(t.Dict[bytes, bytes], ori_headers)
-        ori_headers.update(dict(headers))
-        scope["headers"] = list(ori_headers.items())
+        if isinstance(data, UploadFile):
+            ori_headers = dict(scope.get("headers", dict()))
+            ori_headers = t.cast(t.Dict[bytes, bytes], ori_headers)
+            ori_headers.update(dict(data.headers))
+            scope["headers"] = list(ori_headers.items())
+        if "headers" not in scope:
+            scope["headers"] = []
+
         req = Request(scope)
         req._form = FormData([(field_name, data)])  # type: ignore (using internal starlette APIs)
-        if isinstance(data, bytes):
-            req._body = data
-        else:
+        if isinstance(data, str):
+            req._body = bytes(data, "utf-8")
```

Comment:
I've hard-coded this to be utf-8 for now; I think we might also want to set the encoding header, but I'm going to leave this as-is for now.